libc/src/stdlib/qsort_util.h

```diff
@@ -15,9 +15,13 @@
 #define LIBC_QSORT_QUICK_SORT 1
 #define LIBC_QSORT_HEAP_SORT 2
 
+#ifdef LIBC_OPTIMIZE_FOR_SIZE
+#define LIBC_QSORT_IMPL LIBC_QSORT_HEAP_SORT
+#else
```

Comment:
There doesn't need to be a specific check here. The algorithm selection is done in [`config.json`](https://github.com/llvm/llvm-project/blob/main/libc/config/config.json#L96)

Reply:
I need more information here, my idea was to keep the existing `LIBC_CONF_QSORT_IMPL` flag for compatibility reasons and add a new one that could be re-used across different areas of the library. It's not immediately clear to me how to do that with `config.json` and what the implications would be, and even then it would have to be used here in this way also, no?

Reply:
I agree let's pull it out.